amazonclouddrive/amazonclouddrive.go

```diff
@@ -862,6 +966,50 @@ func (o *Object) Remove() error {
 	return err
 }
 
+// Restore an object
+func (o *Object) Restore() error {
+	return o.fs.pacer.Call(func() (bool, error) {
+		resp, err := o.info.Restore()
+		return o.fs.shouldRetry(resp, err)
+	})
+}
+
+// Changes name of given object
+func (o *Object) Rename(newName string) error {
+	return o.fs.pacer.Call(func() (bool, error) {
+		resp, err := o.info.Rename(newName)
+		return o.fs.shouldRetry(resp, err)
+	})
+}
+
+// Replaces one parent with another, effectively moving the file. Leaves other
+// parents untouched. ReplaceParent cannot be used when the file is trashed.
+func (o *Object) ReplaceParent(oldParentId string, newParentId string) error {
+	fs.Debug(o, "trying parent replace: %s -> %s", oldParentId, newParentId)
+
+	return o.fs.pacer.Call(func() (bool, error) {
+		resp, err := o.info.ReplaceParent(oldParentId, newParentId)
+		return o.fs.shouldRetry(resp, err)
+	})
+}
+
+// Adds one additional parent to object.
+func (o *Object) AddParent(newParentId string) error {
+	return o.fs.pacer.Call(func() (bool, error) {
+		resp, err := o.info.AddParent(newParentId)
+		return o.fs.shouldRetry(resp, err)
+	})
+}
+
+// Remove given parent from object, leaving the other possible
+// parents untouched. Object can end up having no parents.
+func (o *Object) RemoveParent(parentId string) error {
```

Comment:
This should have a small initial letter so it isn't part of the public API of this package.